ipq40xx: ipq4018: add support for Corewav M018U0 SOM S1

neverhover opened the pull request:
ipq40xx: ipq4018: add support for Corewav M018U0 SOM S1

    SoC:    Qualcomm ipq4018
    CPU:    Quad core ARM v7 Cortex A7 717MHz
    RAM:    256MB
    FLASH:  NOR:16MB
    ETH:    5 x GMAC Gigabit
    POE:    802.3 af POE
    WIFI:   1 x 2.4Ghz Atheros qca4019 2x2
            1 x 5 Ghz Atheros qca4019 2x2 MU-MIMO
    USB:    USB 3.0 X 1
    PCI:    1 x mini pcie slot(Only for Lte module with USB2.0)
    SIM:    1 x slot
    SD:     1 x microSD slot
    BTN:    reset
    UART:   1 x 3 pin debug connector (UART)
    GPIOs:  7 pin connector
    POWER:  12V 2A  dc jack

Installation  via bootloader
------------

tftpboot openwrt-ipq40xx-generic-corewav_m018u0-s1-squashfs-sysupgrade.bin
sf probe && sf erase 0x180000 +$filesize && sf write $fileaddr 0x180000 $filesize && reset;

Signed-off-by: Will Zhang <will_zhang@rockeetech.com>

adrianschmutzler commented on target/linux/ipq40xx/files-4.19/arch/arm/boot/dts/qcom-ipq4018-corewav-m018u0.dtsi:
```diff
@@ -0,0 +1,165 @@
+#include <dt-bindings/soc/qcom,tcsr.h>
+
+/ {
+	model = "Corewav-M018U0";
```
Remove this line for the DTSI.

adrianschmutzler commented on target/linux/ipq40xx/files-4.19/arch/arm/boot/dts/qcom-ipq4018-corewav-m018u0.dtsi:
```diff
@@ -0,0 +1,165 @@
+		};
+
+		rng@22000 {
+			status = "ok";
```
Is it still allowed to use "ok" instead of "okay" here?

adrianschmutzler commented on target/linux/ipq40xx/files-4.19/arch/arm/boot/dts/qcom-ipq4018-corewav-m018u0.dtsi:
```diff
@@ -0,0 +1,165 @@
+		};
+
+		blsp_dma: dma@7884000 {
+			status = "ok";
```
ditto.

adrianschmutzler commented on target/linux/ipq40xx/files-4.19/arch/arm/boot/dts/qcom-ipq4018-corewav-m018u0.dtsi:
```diff
@@ -0,0 +1,165 @@
+			status = "ok";
+		};
+	};
+};
```
Add empty line below.

adrianschmutzler commented on target/linux/ipq40xx/image/Makefile:
```diff
@@ -216,6 +216,22 @@ define Device/compex_wpj428
+	DEVICE_MODEL := M018U0
+	DEVICE_VARIANT := S1
+	BOARD_NAME := m018u0-s1
+	DEVICE_DTS := qcom-ipq4018-corewav-m018u0-s1
```
Please have a look at recently merged commit 9c7025b7794ae06774632499547f78d827d13499.
Implications for you:
- Rename DTS(I) files by removing "corewav-"
- Remove DEVICE_DTS here and add `SOC := qcom-ipq4018`
- Update 901-arm-boot-add-dts-files.patch and DTS include lines.

neverhover replied:
Thanks for your review. I changed these files

adrianschmutzler replied:
Looks good as far as these changes are concerned.

blogic commented:
there seems to be a merge commit and the patch needs a rebase

neverhover commented:
> there seems to be a merge commit and the patch needs a rebase

Thanks for your review and rebase done

neverhover commented:
Any other questions? could it be merged? @blogic

adrianschmutzler commented:
"ipq40xx: ipq4018: Add new device Corewav M018U0 SOM refrence design S1"

Add->add
refrence->reference
"new device" -> support for

What does SOM mean?

adrianschmutzler commented:
Maybe remove "reference design" from commit title to make it shorter

adrianschmutzler commented on target/linux/ipq40xx/files-4.19/arch/arm/boot/dts/qcom-ipq4018-m018u0-s1.dts:
```diff
@@ -0,0 +1,76 @@
+#include "qcom-ipq4018-m018u0.dtsi"
+
+/ {
+	model = "Corewav-M018U0-S1";
```
Why not "Corewav M018U0 SOM reference design S1"?

neverhover replied:
Changed

adrianschmutzler commented on target/linux/ipq40xx/files-4.19/arch/arm/boot/dts/qcom-ipq4018-m018u0-s1.dts:
```diff
@@ -0,0 +1,76 @@
+			label = "SBL1";
+			reg = <0x0 0x40000>;
+			read-only;
+		};
```
Add empty lines between nodes as you do in the DTSI.

neverhover replied:
Changed

adrianschmutzler commented on target/linux/ipq40xx/image/Makefile:
```diff
@@ -249,6 +249,22 @@ define Device/compex_wpj428
+	DEVICE_VENDOR := Corewav
+	DEVICE_MODEL := M018U0
+	DEVICE_VARIANT := S1
+	BOARD_NAME := m018u0-s1
```
What's that for?

neverhover replied:
I just refer to other people's writing (eg. dk01.1-c1, qxwlan_e2600ac-c1)

adrianschmutzler replied:
Well, then remove it and check whether it's still working.

neverhover replied:
I removed "BOARD_NAME" and just found #2790 .

adrianschmutzler commented:
Please also add files for kernel 5.4.

neverhover commented:
> "ipq40xx: ipq4018: Add new device Corewav M018U0 SOM refrence design S1"
> 
> Add->add
> refrence->reference
> "new device" -> support for
> 
> What does SOM mean?

SOM is  system on module

neverhover commented:
@adrianschmutzler 5.4 support added

adrianschmutzler commented on target/linux/ipq40xx/base-files/lib/preinit/05_set_iface_mac_ipq40xx.sh:
```diff
@@ -9,6 +9,11 @@ preinit_set_mac_address() {
+	corewav,m018u0-s1)
+		base_mac=$(mtd_get_mac_binary ART 0x1006)
+		ip link set dev eth0 address $(macaddr_add "$base_mac" +2)
+		ip link set dev eth1 address $(macaddr_add "$base_mac" +3)
```
Have you checked whether there are matching addresses in ART 0x0 and/or 0x6? If yes, using them is preferred, if no, just keep it as it is.

neverhover replied:
No i need keep it, because of ART 0x0 ~ 0x1000 are 0xff

adrianschmutzler replied:
Okay.

adrianschmutzler commented on target/linux/ipq40xx/files-4.19/arch/arm/boot/dts/qcom-ipq4018-m018u0.dtsi:
```diff
@@ -0,0 +1,165 @@
+				pinmux {
+					function = "blsp_spi0";
+					pins = "gpio55", "gpio56", "gpio57";
+				};
```
You might add empty lines between nodes here as well.

neverhover replied:
Changed

adrianschmutzler commented on target/linux/ipq40xx/files-5.4/arch/arm/boot/dts/qcom-ipq4018-m018u0.dtsi:
```diff
@@ -0,0 +1,165 @@
+				pinmux {
+					function = "blsp_spi0";
+					pins = "gpio55", "gpio56", "gpio57";
+				};
```
Empty lines...

neverhover replied:
Changed

adrianschmutzler commented:
Two additional comments above.

Despite, I would add "Specifications" as headline before the specifications, and adjust the underline for the flashing instructions to match the entire text.

neverhover committed:
ipq40xx: ipq4018: add support for Corewav M018U0 SOM S1
    SoC:    Qualcomm ipq4018
    CPU:    Quad core ARM v7 Cortex A7 717MHz
    RAM:    256MB
    FLASH:  NOR:16MB
    ETH:    5 x GMAC Gigabit
    POE:    802.3 af POE
    WIFI:   1 x 2.4Ghz Atheros qca4019 2x2
            1 x 5 Ghz Atheros qca4019 2x2 MU-MIMO
    USB:    USB 3.0 X 1
    PCI:    1 x mini pcie slot(Only for Lte module with USB2.0)
    SIM:    1 x slot
    SD:     1 x microSD slot
    BTN:    reset
    UART:   1 x 3 pin debug connector (UART)
    GPIOs:  7 pin connector
    POWER:  12V 2A  dc jack

Installation  via bootloader
------------

tftpboot openwrt-ipq40xx-generic-corewav_m018u0-s1-squashfs-sysupgrade.bin
sf probe && sf erase 0x180000 +$filesize && sf write $fileaddr 0x180000 $filesize && reset;

Signed-off-by: Will Zhang <will_zhang@rockeetech.com>

neverhover commented:
@adrianschmutzler Thanks for your review .I've updated that.
By the way, i found the node 'spi_0_pins, blsp1_spi1' in some dts/dtsi (eg. habanero-dvk.dts) are different of the others(eg. a42.dts, eap2200.dts), some  have empty lines and others are not.
It makes me feel confused...

adrianschmutzler commented:
> some have empty lines and others are not.

The formatting stuff is relatively subjective, and several reviewers do not care at all.

neverhover commented:
I like your style of doing things. so what should i do next.

adrianschmutzler commented:
> Despite, I would add "Specifications" as headline before the specifications in the commit message, and adjust the underline for the flashing instructions to match the entire text.

If you like, also apply the cosmetic change stated above.

I cannot properly review the content on this target, so with the formal stuff sorted out you may wait for someone to further review/merge.

neverhover commented:
I've updated the commit and applied the cosmetic change stated above.

ynezz commented on target/linux/ipq40xx/files-5.4/arch/arm/boot/dts/qcom-ipq4018-m018u0-s1.dts:
```diff
@@ -0,0 +1,86 @@
+};
+
+&blsp1_spi1 {
+	mx25l25635f@0 {
```
@neverhover Can you please change the node name to more generic name `flash@0` and try if it boots for you?

ynezz commented on target/linux/ipq40xx/files-4.19/arch/arm/boot/dts/qcom-ipq4018-m018u0-s1.dts:
```diff
@@ -0,0 +1,86 @@
+};
+
+&blsp1_spi1 {
+	mx25l25635f@0 {
```
@neverhover Can you please change the node name to more generic name `flash@0` and try if it boots for you?

ynezz reviewed (changes requested):
Please fix the last nitpicks and rebase.

adrianschmutzler commented:
@neverhover Ping.

adrianschmutzler commented:
DTS files for ipq40xx have recently been merged in files instead of files-5.4 and files-4.19. So, unless your files are different, just do the same, please.

adschm commented:
Unfortunately, this seems abandoned, although there weren't much changes remaining. So, I will close it now. Please ask for reopening if you want to continue working on it (_before_ pushing anything to the branch).
